### PR TITLE
[Feature #5723] Output UDP stream (for multicast)

### PR DESCRIPTION
**Pull request by innes-labs**
Hi Jaroslav,
Waiting for your review ...

Stéphane Duperron

### CODE REVIEW

File: src/wrappers.c [hunk 1]
```diff
@@ -68,9 +68,17 @@ tvh_write(int fd, const void *buf, size_t len)
 {
   int64_t limit = mclk() + sec2mono(25);
   ssize_t c;
+  ssize_t towrite;
+  int txsize = len;
+  int sotype = SOCK_STREAM;
+  socklen_t sz = sizeof(sotype);
+  if (getsockopt(fd, SOL_SOCKET, SO_TYPE, &sotype, &sz) == -1 || sotype == SOCK_DGRAM) {
+    txsize = 188*7; // under MTU size for udp sockets
```
Comment (Trujulu) [inline src/wrappers.c]: Isn't better to set it externally, like src/input/mpegts/satip/satip_frontend.c#L1572 ?

**Comment by Flole998** [comment]
Also could you squash your commits together please, that makes reviewing it a lot easier.

**Comment by Flole998** [comment]
I just noticed that you changed the intendation at various places in the webui.c file so it's not correct anymore, would be great if you could fix that aswell

**Comment by Saentist** [comment]
@innes-labs can you update pull request?

**Comment by innes-labs** [comment]
I try to do the update in the weeks

**Comment by Saentist** [comment]
Hope this time will be implemented.

**Comment by innes-labs** [comment]
Pull request has been updated. Waiting for your review guys

**Comment by Saentist** [comment]
Again will request pull request.
Please anyone with can help with optimisation to this PR, to contribute and comment.

**Comment by Flole998** [comment]
I just tested this and it seems like when the UDP Port is closed and a ICMP response is sent the sending of the traffic is stopped but the subscription is staying active. Either the subscription should be removed with an error or there should be an attempt after some time to start transmitting packets again. There is no real point in having a subscription that never forwards any traffic.

**Comment by Flole998** [comment]
@innes-labs Could you please address the issue I mentioned above? Other than that it looks good so it should be good to be merged then.

**Comment by innes-labs** [comment]
@Flole998 In your comment on 1 Mar, do you mean ICMP or IGMP ?

**Comment by Flole998** [comment]
ICMP, when you try to send data to a closed port there is an ICMP response indicating that. Write will fail with an error in that case iirc.

**Comment by innes-labs** [comment]
@Flole998 How did you close the UDP port in your test ? I failed to generate the ICMP message (type 3 code 3 ?) you mentioned. But I think that the last commit will fix. Could you retry ?

**Comment by Flole998** [comment]
Please squash commits together.

I used netcat and simply stopped it after some time. I will test the new commit.

**Pull request merged**